packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -160,6 +160,15 @@ abstract class Route<T> {
     });
   }
 
+  /// Called after route is added to navigator.
+  @protected
+  @mustCallSuper
+  TickerFuture didAdd() {
```

Comment:
We need the adding state for pages. Imagine you add three pages to the list of pages. The topmost should be pushed, the other two will just be added after the push of the topmost page is compelete to avoid visual glitches. THat's what we need the adding state for.